File: SharedKernel/CRM.SharedKernel.API/Responses/BaseResponse.cs
namespace CRM.SharedKernel.API.Responses;

public class BaseResponse<T> : BaseResponse
{
	public T Data { get; set; }
}

public class BaseResponse
{
	public ResponseStatus Status { get; set; } = default!;
	public Dictionary<string, string> Errors { get; set; } = [];
}

public class ResponseStatus
{
	public int Code { get; set; }
	public string Message { get; set; }

	public ResponseStatus(int code = 1, string message = "Unexpected Error Occurred!")
	{
		Code = code;
		Message = message;
	}

	public ResponseStatus(bool status, string message = "Unexpected Error Occurred!")
	{
		Code = status ? 0 : 1;
		Message = message;
	}
}

File: Modules/Users/Modules.Users.UI/src/app/shared/interfaces/base-response.ts
import { ResponseStatusEnum } from '@shared/Enums/response-status-enum';

export interface BaseResponse<T> {
  data: T;
  status: responseStatus;
  totalItemsCount: number;
  // Backend may also return plain PaginationResponse or DTO directly on 200
  // This keeps backward compat for ApiService that now unwraps both shapes
}

export interface responseStatus {
  code: ResponseStatusEnum;
  message: string;
}

// Backend PaginationResponse — matches CRM.SharedKernel PaginationResponse<T>
export interface PaginationResponse<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalItemsCount: number;
}

// Backend error envelope — ToMVCProblem() -> BaseResponse { Status, Errors }
export interface ApiErrorResponse {
  status: { code: number; message: string };
  errors: Record<string, string>;
}
